solana-programs/staking-bridge/tests/staking-bridge.ts

```diff
@@ -1,16 +1,293 @@
+      quoteMint: market.quoteMint
+    })
+    console.log("amm poolKeys: ", JSON.stringify(poolKeys))
+
+    const { vaultOwner, vaultNonce } = await getVaultOwnerAndNonce(serumMarketPublicKey, serumDexProgram)
+    if (vaultNonce.toNumber() != market.vaultSignerNonce) {
+      console.log(
+        "withdraw vaultOwner:",
+        vaultOwner.toString(),
+        "vaultNonce: ",
+        vaultNonce.toNumber(),
+        "market.vaultSignerNonce:",
+        market.vaultSignerNonce.toString()
+      )
+      throw ("vaultSignerNonce incorrect!")
+    }
+
+    // Get associated token accounts for the staking bridge PDA.
+    // Create them if they don't exist.
+    const usdcTokenAccount = await getOrCreateAssociatedTokenAccount(
+      connection,
+      feePayerKeypair,
+      SOL_USDC_TOKEN_ADDRESS_KEY,
+      stakingBridgePda,
+      true // allowOwnerOffCurve: we need this since the owner is a program
+    )
+    const audioTokenAccount = await getOrCreateAssociatedTokenAccount(
+      connection,
+      feePayerKeypair,
+      SOL_AUDIO_TOKEN_ADDRESS_KEY,
+      stakingBridgePda,
+      true // allowOwnerOffCurve: we need this since the owner is a program
+    )
+
+    // Amount of SOL USDC to be swapped for a minimum amount of SOL AUDIO expected to be received from the swap.
+    const wholeAmountIn = 0.001
+    const { amountIn, minimumAmountOut } = getMinimumAmountOutFromAmountIn(wholeAmountIn);
+
+    const accounts = {
+      programId: poolKeys.programId,
+      amm: poolKeys.id,
+      ammAuthority: poolKeys.authority,
+      ammOpenOrders: poolKeys.openOrders,
+      ammTargetOrders: poolKeys.targetOrders,
+      poolCoinTokenAccount: poolKeys.baseVault,
+      poolPcTokenAccount: poolKeys.quoteVault,
+      serumProgram: serumDexProgram,
+      serumMarket: serumMarketPublicKey,
+      serumBids: market.bids,
+      serumAsks: market.asks,
+      serumEventQueue: market.eventQueue,
+      serumCoinVaultAccount: market.baseVault,
+      serumPcVaultAccount: market.quoteVault,
+      serumVaultSigner: vaultOwner,
+      userSourceTokenAccount: usdcTokenAccount.address,
+      userDestinationTokenAccount: audioTokenAccount.address,
+      userSourceOwner: stakingBridgePda,
+      splTokenProgram: TOKEN_PROGRAM_ID,
+    }
+    console.log('Raydium swap accounts:', { accounts })
+
     // Add your test here.
-    const tx = await program.methods.initialize().rpc();
+    const tx = await program.methods
+      .raydiumSwap(
+        amountIn,
+        minimumAmountOut,
+        vaultNonce,
+        stakingBridgePdaBump,
+      )
+      .accounts(accounts)
+      .rpc();
     console.log("Your transaction signature", tx);
-  });
-});
+  })
+
+  it('posts the wormhole token bridge transfer message', async () => {
+    const messageKeypair = Keypair.generate()
+    const messagePublicKey = messageKeypair.publicKey
+
+    // How many SOL AUDIO tokens to convert into ETH AUDIO tokens
+    const wholeAmount = 0.00001
```

Comment:
wormholeAmount? this is a confusing name. maybe you mean entireAmount?

Reply:
it says `wholeAmount`, not `wormholeAmount`

Reply:
yeah i'm saying "whole" is a hard word to grok because it looks like wormhole versus entire, assuming you meant that ;)

Reply:
ah. yeah i see. i can call it `entire`. what im going for is like the non-decimal-unit. as in `1` SOL instead of the decimals which would be `1000000000` lamports. not sure if `entire` captures that but down to rename

Reply:
oh i didn't understand that

Reply:
`wholeNumberAudioAmount`?

Reply:
I've seen patterns that call this `uiAmount`